2025/build-tooling/demo/final/src/vite-env.d.ts

```diff
@@ -0,0 +1,3 @@
+/// <reference types="vite/client" />
+/// <reference types="@esri/calcite-components/types/react" />
+/// <reference types="@arcgis/map-components/types/react" />
```

Comment:
I think this should just be a generic declarations file, referenced in the tsconfig. This way typescript picks it up, rather than making it specific to Vite. Eg...

```ts
/** FILE: /src/declarations.d.ts **/

/// <reference types="vite/client" />
import "@esri/calcite-components/dist/types/react";
import "@arcgis/map-components/types/react";

/** FILE: tsconfig.json **/

{
  // ... rest
  "include": ["src/declarations.d.ts"]
}
```

Reply:
> referenced in the tsconfig

doesn't need to be referenced in tsconfig if it is included in the "include" tsconfig pattern

> import "@esri/calcite-components/types/react";
> import "@arcgis/map-components/types/react";

These files are type-only - they don't exist at runtime
While it should be fine in a .d.ts file, they would cause build error in a .ts file
Keeping them as /// <reference> has the benefit of keeping our instructions simpler: "just drop them into any TypeScript file in your project, for example, src/vite-env.d.ts" rather than "create a .d.ts file if you don't already have one and put these imports inside"

Reply:
Got it, works for me!